### PR TITLE
Fix onboarding woes

### PR DESCRIPTION
**Pull request by codeluggage**
This fixes an issue with permissions for the unique onboarding channels that are created when a new member joins the server. It broke between https://github.com/scrimba/discord-bot/commit/9a4d9ce4b55d544daf44afdb08956e2c32e60a11 and now.

### CODE REVIEW

File: src/onboarding.imba [hunk 1]
```diff
@@ -177,16 +177,22 @@ const build_bot = do(bot)
 	const on_add = do(member)
 		const name = "{WELCOME_PREFIX}{member.user.username}_{member.user.id}"
 		const channel = await member.guild.channels.create name,
+			type: 'GUILD_TEXT'
```
Comment (Pistus) [inline src/onboarding.imba:180]: sets it as text channel?

File: src/onboarding.imba [hunk 2]
```diff
@@ -177,16 +177,22 @@ const build_bot = do(bot)
...
 			parent: ONBOARDING_CATEGORY_ID
 			permissionsOverwrites: [
-				{ id: member.id, allow: VIEW_PERMISSION }
-				{ id: EVERYONE_ROLE_ID, deny: VIEW_PERMISSION }
-				{ id: bot.user.id, allow: VIEW_PERMISSION }
+				{ id: EVERYONE_ROLE_ID, deny: [Permissions.FLAGS.VIEW_CHANNEL] }
```
Comment (Pistus) [inline src/onboarding.imba:183]: Did using the Permissions.* change anything by itself?

File: src/onboarding.imba (hunk 1 above)
Comment (codeluggage) [reply src/onboarding.imba:180]: Yes. It is the default but doesn't hurt to be explicit.

File: src/onboarding.imba (hunk 2 above)
Comment (codeluggage) [reply src/onboarding.imba:183]: No. They refer back to string values, but it does give us type safety / clumsy finger safety.

File: package.json [hunk 3]
```diff
@@ -28,5 +28,8 @@
 		"knex": "^0.95.12",
 		"pg": "^8.7.1",
 		"postgres": "^1.0.2"
+	},
+	"engines": {
+		"node": ">=16.6.2"
```
Comment (haikyuu) [inline package.json:33]: Nice addition

**Review by haikyuu — COMMENTED** [review]
Looks Good. Glad it was fixed 👍🏽

**Pull request merged**